packages/examples/src/simple-chat.tsx

```diff
@@ -10,4 +13,157 @@ function App() {
+      }
+      done = true;
+      resolveFinalResult!(lastFrame);
+    })();
+
+    // Function to create a new ReadableStream for each consumer
+    function createFrameStream() {
+      const streamController = new TransformStream();
+      const writer = streamController.writable.getWriter();
+      function handleFrame(frame: string) {
+        writer.write(frame);
+      }
+      handleFrame(lastFrame);
+      eventEmitter.on('frame', handleFrame);
+
+      // Close the stream when done
+      if (done) {
+        eventEmitter.off('frame', handleFrame);
+        writer.close();
+      } else {
+        finalResult.then(() => {
+          writer.write(lastFrame);
+          eventEmitter.off('frame', handleFrame);
+          writer.close();
+        });
+      }
+
+      return streamController.readable;
+    }
+
+    function createDeltaStream() {
+      let lastEmittedValue = '';
+      const transformStream = new TransformStream({
+        transform(latestFrame, controller) {
+          const delta = latestFrame.slice(lastEmittedValue.length);
+          lastEmittedValue = latestFrame;
+          controller.enqueue(delta);
+        },
+      });
+      return createFrameStream().pipeThrough(transformStream);
+    }
+
+    return {
+      createFrameStream,
+      createDeltaStream,
+      finalResult,
+    };
+  }
+
+  const renderContext = AI.createRenderContext(/* take logger */);
+  const memoized = renderContext.memo(renderable);
+  const treeStreamRender = renderContext.render(memoized);
+  const appendStreamRender = renderContext.render(memoized, { appendOnly: true });
+  const treeStreamRenderAdapter = makeRenderAdapter(treeStreamRender);
+
+  // treeStreamRenderAdapter.finalResult.then(out => console.log('out result', out))
+
+  const appendStreamRenderAdapter = makeRenderAdapter(appendStreamRender);
+
+  return {
+    treeStream: treeStreamRenderAdapter.createFrameStream,
+    appendStream: appendStreamRenderAdapter.createFrameStream,
+    deltaStream: treeStreamRenderAdapter.createDeltaStream,
+    // We could pull the final result from either adapter – they're equivalent.
+    result: appendStreamRenderAdapter.finalResult,
+
+    writeToStdout: () =>
+      // @ts-expect-error
+      new ReadableWebStreamToNodeStream.ReadableWebToNodeStream(appendStreamRenderAdapter.createDeltaStream()).pipe(process.stdout),
+  };
+}
+
+/**
+ * Helper function for debugging.
+ */
+async function streamToValues(stream: ReadableStream) {
+  const values: any[] = [];
+
+  await stream.pipeTo(
+    new WritableStream({
+      write(chunk) {
+        values.push(chunk);
+      },
+    })
+  );
+
+  return values;
+}
+
+const renderable = <App />;
+const rendered = render(renderable);
+
+console.log('=== First Render ===');
+// You can consume a stream multiple times.
+console.log(
+  await Promise.all([
+    streamToValues(rendered.treeStream()),
+    streamToValues(rendered.treeStream()),
+    streamToValues(rendered.deltaStream()),
+  ])
```

Comment:
What's the scenario where one wants to consume the same stream in multiple formats/multiple times? That feels like it should inform what the right semantics are, i.e. should they operate independently or in sync?